### PR TITLE
Pages error loading

### PR DESCRIPTION
**Pull request by haitham-akram**
this PR contains a 404 error page, loading and use context for auth and categories data

### CODE REVIEW

File: client/vite.config.ts.timestamp-1687784474324-6a60defc7afd.mjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// vite.config.ts
+import { defineConfig } from "file:///E:/GSG/Project/Petopia/client/node_modules/vite/dist/node/index.js";
+import react from "file:///E:/GSG/Project/Petopia/client/node_modules/@vitejs/plugin-react/dist/index.mjs";
```
Comment (Mu7ammadAbed) [inline client/vite.config.ts.timestamp-1687784474324-6a60defc7afd.mjs:3]: What are these ?

File: package.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
     "cors": "^2.8.5",
     "dotenv": "^16.1.4",
     "express": "^4.18.2",
+    "iconsax-react": "^0.0.8",
```
Comment (Mu7ammadAbed) [inline package.json:36]: this belongs to the react package json

File: client/src/index.tsx [hunk 3]
```diff
@@ -11,6 +12,10 @@ const router = createBrowserRouter([
     path: '/',
     element: <App />,
   },
+  {
+    path: '*',
+    element: <PageNotFound />,
+  },
```
Comment (Mu7ammadAbed) [inline client/src/index.tsx:18]:
```suggestion
const router = createBrowserRouter([
  {
    path: '/',
    element: <App />,
    errorElement: <PageNotFound />,
  },
  {
    path: '/post',
    element: <PostCard post={post} />,
  },
]);
```
and move the router to a separate file

File: client/src/components/AuthContext.tsx
Comment (Mu7ammadAbed) [inline client/src/components/AuthContext.tsx:1]: You created a `context` folder inside the `components` folder you should move this file inside it

File: client/src/components/commons/LoadingComponent.tsx
Comment (Mu7ammadAbed) [inline client/src/components/commons/LoadingComponent.tsx:1]: We might need to edit this Components' style in the future, but for now we will keep it just like this

File: client/src/index.css [hunk 4]
```diff
@@ -19,7 +19,35 @@
...
   }
-  
+
   body {
     font-family: var(--main-font);
-  }
\ No newline at end of file
+    margin: 0;
+  }
+.loading-container{
```
Comment (Mu7ammadAbed) [inline client/src/index.css:27]: These css lines shouldn't be here in the root `index.css`, they belong to another place

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/vite.config.ts.timestamp-1687784474324-6a60defc7afd.mjs (hunk 1 above)
Comment (haitham-akram) [reply client/vite.config.ts.timestamp-1687784474324-6a60defc7afd.mjs:3]: this file TSC created it and I didn't notice it until now

File: package.json (hunk 2 above)
Comment (haitham-akram) [reply package.json:36]: it was there by mistake

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**